confgenerator/resourcedetector/bms_detector.go

```diff
@@ -62,6 +62,21 @@ func (r BMSResource) MonitoredResource() *monitoredres.MonitoredResource {
 	}
 }
 
+func (r BMSResource) PrometheusStyleMetadata() map[string]string {
+	metaLabels := map[string]string{
+		"__meta_bms_instance_id": r.InstanceID,
+		"__meta_bms_project":     r.Project,
+		"__meta_bms_location":    r.Location,
+	}
+	// Set the location, namespace and cluster labels.
+	metaLabels["location"] = r.Location
+	metaLabels["namespace"] = r.InstanceID
+	metaLabels["cluster"] = "__bms__"
+	// Set some curated labels.
+	metaLabels["instance_name"] = r.InstanceID
```

Comment:
Would this not be surprising to the user? If not, why not set `__meta_bms_instance_name` accordingly?

Reply:
Are you suggesting that the instance_name should not be set to r.InstanceID? I realize that I simply copied this from gce_detector.go without considering its implications. I agree that it might be surprising to the user. I've removed this line to avoid any unexpected behavior.